lib/aggregate.js

```diff
@@ -1105,14 +1105,9 @@ function prepareDiscriminatorPipeline(aggregate) {
     const discriminatorKey = discriminatorMapping.key;
     const discriminatorValue = discriminatorMapping.value;
 
-    // If the first pipeline stage is a match and it doesn't specify a `__t`
-    // key, add the discriminator key to it. This allows for potential
-    // aggregation query optimizations not to be disturbed by this feature.
-    if (originalPipeline[0] && originalPipeline[0].$match && !originalPipeline[0].$match[discriminatorKey]) {
```

Comment:
Please undo these changes, this code shouldn't be removed. No need to change `aggregate.js` at all, that's outside the scope.